src/Mod/Sketcher/Gui/TaskSketcherMessages.h

```diff
@@ -39,28 +39,26 @@ namespace SketcherGui {
 
 class ViewProviderSketch;
 
-class TaskSketcherMessages : public Gui::TaskView::TaskBox
+class TaskSketcherMessages : public QWidget
 {
     Q_OBJECT
 
 public:
-    explicit TaskSketcherMessages(ViewProviderSketch *sketchView);
+    explicit TaskSketcherMessages(ViewProviderSketch* sketchView, QWidget* parent = nullptr);

```

Comment:
I am unsure why have you done this.

At first I thought you had realised that now TaskSketcherMessages is a QWidget. You know that Widgets are generally added to the parent in QT's memory model and then the parent is responsible for deleting them. So you added this field.

However, in the cpp file you are instantiating the object with only one parameter (without parent). So it has no effect.

Then, I start to think maybe you are doing it for a different reason. But I cannot find one, as passing a nullptr as parent is the same as passing nothing at all (a QWidget's default constructor has parent defaulted to nullptr).

In any case:

Observation 1: If you did this to use the QT's memory management, the constructor is taking a default argument nullptr. If the parent is not passed, the object is created, but who is responsible for the deletion? Nobody.

Observation 2: A TaskBox is a widget, but a TaskBox does not pass the parent as parameter. If this is working fine, the TaskBox must have a different memory  model (which may not follow the QT's memory model). If we look into that, the widgets of a TaskDialog are added to the Contents array of the TaskDialog class. If you look into the destructor of TaskDialog, you will see that the pointers inside Contents are destroyed there. So TaskDialog has a different memory model in place.

Observation 3: Because the TaskSketcherMessages object is added to the Contents array, you potentially have two memory models running in parallel (the one of QT via a parent being different from nullptr, and the one provided by TaskDialog). 

I tend to think that adding this parameter here is highly misleading for future developers, as they are invited to think the object may be used in a way that is not intended.

Is there any reason I have not considered for which this should stay?

Reply:
For this I reused the same as I used in previous PR. Now why I did that in the first place last october I'm not sure. I think it's because I copied the SketcherGeneralWidget as a base for this solver widget, and mixed it with the things that taskBox does not pass the parent. So it's an innaccuracy, I will remove it.

Reply:
Thanks for fixing it.

Try to always read the code before PR-ing it and questioning what looks odd. Aside from possibly fixing all mistakes, it will help you review code of others. Eventually you will need to do that too.